src/Controls/src/Core/HandlerImpl/VisualElement/VisualElement.Impl.cs

```diff
@@ -214,5 +236,35 @@ static void ValidatePositive(double value, string name)
 		}
 
 		Thickness IView.Margin => Thickness.Zero;
+
+		void NotifyShadowChanges()
+		{
+			if (Shadow != null)
+			{
+				Shadow.Parent = this;
```

Comment:
Not sure if maybe should check if Parent is already assigned (not null).

Reply:
If we are moving shadow from one place to the parent would be this one